File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        # maintain a stack consisting of temp and item, go through the array
        # and add to stack if it is empty or if the element before it is >
        # if current element greater than the element before it, pop from stack and keep track of count 
        # until prev element is >= current element, add this element to stack
        res = [0] * len(temperatures)
        stack = []

        for i in range(len(temperatures)):
            while len(stack) > 0 and stack[-1][0] < temperatures[i]:
                prev = stack.pop()
                res[prev[1]] = i - prev[1]
            stack.append([temperatures[i], i])
        
        return res



        

File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> res;

        for (int i = 0; i <= n; i++) {
            int mask = 1;
            int count = 0;

            for (int j = 0; j < 32; j++) {
                if (mask & i) {
                    count++;
                }
                mask = mask << 1;
            }

            res.push_back(count);
        }

        return res;
    }   
};
